revision (#4)
yoyololicon/mdx-submissions21: paper.md

In this work, we improved upon the state-of-the-art (SoTA) models and successfully combined the best of both worlds.
The backbones of the proposed framework, dubbed Danna-Sep^[<https://github.com/yoyololicon/danna-sep>], are two spectrogram-based models including a modified X-UMX and U-Net, and an enhanced Demucs as the waveform-based model.
Given an input of mixture, we linearly combined respective outputs from the three models to obtain the final result.
We showed in the experiments that, despite its simplicity, Danna-Sep surpassed the SoTA models by a large margin in terms of Source-to-Distortion Ratio. 

# Method

Danna-Sep is a combination of three different models: X-UMX, U-Net, and Demucs.
We describe the enhancements made for each model in the following subsections.

## X-UMX
X-UMX [@sawata20] improved upon UMX[@stoter19] by concatenating hidden layers of UMX to enable sharing information among all target instruments.
We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $j$-th source as follows:

> $$\mathcal{L}_{MSE}^J = \sum_{j=1}^J\sum_{t,f}|Y_j(t, f) - \hat{Y}_j(t, f)|^2$$,

Reviewer: ```suggestion
$$\mathcal{L}_{MSE}^J = \sum_{j=1}^J\sum_{t,f}|Y_j(t, f) - \hat{Y}_j(t, f)|^2$$
```

## Demucs
For Demucs [@defossez2019music], we built upon the variant with 48 hidden channels, and enhanced the model by replacing the decoder with four independent decoders responsible for four respective sources.
Each decoder has the same architecture as the original decoder, except for size of the hidden channel which was reduced to 24. 
This makes the total number of parameters comparable with the original Demucs.
The training loss aggregates the L1-norm between estimated and ground-truth waveforms of the four sources.
The model took approximately 10 days to train on a single RTX 3070 using mixed precision with a batch size of 16, and four steps of gradient accumulation.

## Danna-Sep
In order to obtain the final output of our framework, we calculated weighted average of individual outputs from the above-mentioned models.
Experiments were conducted to search for optimal weighting. The optimal weights for each source, types of input domain (T for waveforms, TF for frequency masking), and the sizes of the models are given in the following table.

> |         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | |---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:| | X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136

Reviewer: ```suggestion
|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | 
|---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:| 
| X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
```
broken table

| U-Net   | 0.2   | 0.17 | 0.5   | 0.4    | TF | 61
| Demucs  | 0.6   | 0.73 | 0.5   | 0.4    | T | 733

All models were trained on the training set of musdb18-hq [@musdb18-hq] using an Adam optmizier[@kingma2014adam]. 

# Separation performances

> We performed the benchmark using musdb18-hq [@musdb18] as the dataset for a fair comparison, and evaluated the models in terms of Signal-to-Distortion Ratio (SDR) [@vincent2006performance] derived from *museval* [@fabian_robert_stoter_2019_3376621].

Reviewer: We trained Danna-Sep on musdb18-hq (high quality) but the benchmark was done on musdb18 (low quality) so the score can be compared to other works. This is different from X-UMX and Demucs, so I add one more line to state that we picked baselines that were also trained on high quality dataset.
I prefer the previous version.

Author: Thanks for clarifying; refer to [commit](https://github.com/yoyololicon/mdx-submissions21/pull/4/commits/09ada99f83b65e567bf7c99a9d85371155d76190) for the update.

## Demucs
For Demucs [@defossez2019music], we built upon the variant with 48 hidden channels, and enhanced the model by replacing the decoder with four independent decoders responsible for four respective sources.
Each decoder has the same architecture as the original decoder, except for size of the hidden channel which was reduced to 24. 
This makes the total number of parameters comparable with the original Demucs.
The training loss aggregates the L1-norm between estimated and ground-truth waveforms of the four sources.
The model took approximately 10 days to train on a single RTX 3070 using mixed precision with a batch size of 16, and four steps of gradient accumulation.

## Danna-Sep
In order to obtain the final output of our framework, we calculated weighted average of individual outputs from the above-mentioned models.
Experiments were conducted to search for optimal weighting. The optimal weights for each source, types of input domain (T for waveforms, TF for frequency masking), and the sizes of the models are given in the following table.

> |         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | 

Reviewer: ```suggestion

|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | 
```

The two types of models have complementary strengths; the former is superior given harmonic sources such as vocals, while the latter demonstrates better results for percussion and bass instruments.
In this work, we improved upon the state-of-the-art (SoTA) models and successfully combined the best of both worlds.
The backbones of the proposed framework, dubbed Danna-Sep^[<https://github.com/yoyololicon/danna-sep>], are two spectrogram-based models including a modified X-UMX and U-Net, and an enhanced Demucs as the waveform-based model.
Given an input of mixture, we linearly combined respective outputs from the three models to obtain the final result.
We showed in the experiments that, despite its simplicity, Danna-Sep surpassed the SoTA models by a large margin in terms of Source-to-Distortion Ratio. 

# Method

Danna-Sep is a combination of three different models: X-UMX, U-Net, and Demucs.
We describe the enhancements made for each model in the following subsections.

## X-UMX
X-UMX [@sawata20] improved upon UMX[@stoter19] by concatenating hidden layers of UMX to enable sharing information among all target instruments.

> We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $j$-th source as follows:

Reviewer: ```suggestion
We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $J$ sources as follows:
```

$$\mathcal{L}_{MSE}^J = \sum_{j=1}^J\sum_{t,f}|Y_j(t, f) - \hat{Y}_j(t, f)|^2$$

> where $Y(t, f)$ and $\hat{Y}(t, f)$ are ground-truth and estimated time-frequency representations, respectively. 

Reviewer: ```suggestion
where $Y_j(t, f)$ and $\hat{Y}_j(t, f)$ are ground-truth and estimated time-frequency representations for the $j$-th source, respectively. 
```
